### PR TITLE
Implement methods to convert between Component and Brigadier's Message

### PR DESCRIPTION
**Pull request by jpenilla**

### CODE REVIEW

**Review by MiniDigger — APPROVED** [review]

File: Paper-MojangAPI/src/main/java/io/papermc/paper/brigadier/PaperBrigadierProvider.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+interface PaperBrigadierProvider {
+    class Holder {
+        private static PaperBrigadierProvider INSTANCE;
+    }
+
+    static @NonNull PaperBrigadierProvider instance() {
+        return requireNonNull(Holder.INSTANCE, "PaperBrigadierProvider has not yet been initialized!");
+    }
+
+    static void initialize(final @Nullable PaperBrigadierProvider instance) {
```
Comment (Proximyst) [inline Paper-MojangAPI/src/main/java/io/papermc/paper/brigadier/PaperBrigadierProvider.java]: Document this should only be initialised by the server. `@Deprecated` would be a completely okay annotation here.

**Review by Proximyst — APPROVED** [review]

File: Paper-MojangAPI/src/main/java/io/papermc/paper/brigadier/PaperBrigadierProvider.java (hunk 1 above)
Comment (MiniDigger) [reply Paper-MojangAPI/src/main/java/io/papermc/paper/brigadier/PaperBrigadierProvider.java]: I mean, the class and method are package private, so doesn't matter that much

Comment (Proximyst) [reply Paper-MojangAPI/src/main/java/io/papermc/paper/brigadier/PaperBrigadierProvider.java]: Didn't even notice 😅

**Review by Proximyst — APPROVED** [review]

**Review by JRoy — APPROVED** [review]
i was paid copious amounts of money for this approval

**Pull request merged**